src/uct/cuda/cuda_ipc/cuda_ipc_ep.c

```diff
@@ -133,15 +114,20 @@ uct_cuda_ipc_post_cuda_async_copy(uct_ep_h tl_ep, uint64_t remote_addr,
         return UCS_OK;
     }
 
-    status = uct_cuda_ipc_map_memhandle(&key->super, key->super.dev_num,
```

Comment:
do we need key->super.dev_num now?

Reply:
looking at the changes now, do you think it would be better/safer to cache relevant dev_num in `uct_cuda_ipc_cache_region_t` instead to avoid passing it in many devices?

Reply:
done